public/app/features/templating/template_srv.ts

```diff
@@ -347,6 +349,31 @@ export class TemplateSrv implements BaseTemplateSrv {
     });
   }
 
+  getVariablesMapInTemplate(
+    target: string,
+    scopedVars: ScopedVars,
+    format?: string | Function
+  ): Record<string, VarValue> {
+    const regexp = new RegExp(this.regex);
+    const values: Record<string, VarValue> = {};
+
+    target.replace(regexp, (match, var1, var2, fmt2, var3, fieldPath, fmt3) => {
+      const variableName = var1 || var2 || var3;
+      const fmt = fmt2 || fmt3 || format;
+      const value = this.getVariableValue(variableName, fieldPath, scopedVars);
+      if (value !== null && value !== undefined) {
+        const variable = this.getVariableAtIndex(variableName);
+        const text = this.getVariableText(variableName, value, scopedVars);
+        values[variableName] = this.formatValue(value, fmt, variable, text);
```

Comment:
Also static links are shown as:

<img width="468" alt="Screenshot 2023-03-07 at 11 02 21" src="https://user-images.githubusercontent.com/745532/223389449-c83d4e4b-960e-478e-a9aa-e99dcb207a66.png">

Maybe we could display the name of field in that case?

Reply:
With the `__data.fields.traceID` example, this led me down a bit of a rabbit hole I wanted to document here to explain some of the changes I made that might seem extraneous but I think will be helpful for anyone who has to mess with this in the future.

Support for defining variables as <variable name>.<path> was added [here](https://github.com/grafana/grafana/pull/18918/files#diff-a9e62983b14551ca35d1072859608fb78f79716010b516cadc4a9574e1f77347L10) in 2019, and the accompanying comment explaining what kinds of variable formats were supported was never changed to reflect the feature. Thus, the comment about the third supported variable format does not match the regex and doesn't comprehensively define what the regex supports ([src](https://github.com/grafana/grafana/blob/main/public/app/features/variables/utils.ts#L23)). I fixed that to be both more accurate and fully explain the regex.

There are 6 capture groups defined in the variable regex, and each of them are accessible in the replace function ([src](https://github.com/grafana/grafana/blob/main/public/app/features/templating/template_srv.ts#L294)). In my opinion, it's not clear that the parameter names of the replace function map to the capture groups as named in the comment about the variable regex, so I added a comment clarifying that.

The thing you'd like added is the fieldPath capture group. I changed it where if a variable with a fieldPath is defined, it will display both the name of the variable and it's field path together - that will match what is in the query.

Reply:
> Also static links are shown as:
Maybe we could display the name of field in that case?

@ifrost How are you adding static links like this one? I just want to make sure I'm replicating what you're doing.

Reply:
Just using a target query with no variables, can be even an empty query.

Reply:
Ahh sorry for the delay, I forgot about this. Here's a look at what a static link now looks like

<img width="653" alt="Screenshot 2023-03-10 at 11 49 07 AM" src="https://user-images.githubusercontent.com/1533128/224387819-e0f1a1a1-78b0-4499-93e5-2d6c5e53e496.png">